### ISSUE TITLE
Question about distillation on MLM task

### ISSUE THREAD

**Issue by vincentlux** [airaria/TextBrewer#17]
Hi,

Thank you so much for open sourcing this toolkit! This is very helpful. I tried to finetune on several downstream tasks and it works great.

May I ask if you ever tried to distill a random-initialized model using a large teacher model finetuned on a specific dataset using MLM as objective? Specifically, I am trying to distill a `T4tiny` using `roberta-wwm-ext` finetuned on my own dataset using MLM as objective. However, the perplexity sticks at around 60... 

Speaking of implementation, my adaptor returns `losses, logits, hidden, attention` for distillation, and the intermediate matches I used is following the T4tiny configuration: `L4t_hidden_mse, L4_hidden_smmd`. The dataset I am using for MLM is part of webtext qa dataset, which sizes ~500mb.

Is it possible that you could provide some notes that I should pay attention to? Thank you so much for your kind help!

Best,
Vincent

**Comment by airaria** [comment]
Thank you for your feedback and appreciation of this work!
But we haven't performed distillation with MLM task. We may do this next month.
To solve your problem, there are some tips/checklist that may help:
1. Make sure that the order of the batch and the order of the arguments in the `model.forward` method are the same
2. Return "logits_mask" in the adaptor, which can be used to avoid distilling on padding positions
3. If you're using Transformers (rather than the old pytorch_pretrained_bert), call the train method of distiller with 
`scheduler_class` and `scheduler_args`, **not** ``scheduler``; and setting `max_grad_norm=1` may help.
4. Use a high temperature (e.g., 8) and high learning rate (e.g. >1e-4)
2. You may refer to the original[ TinyBERT paper ](https://arxiv.org/abs/1909.10351) to see if the hyperparameters there work.

We will update the repo once we have any results on MLM task. 

Thanks again for your feedback! Hope my reply helps.

**Comment by vincentlux** [comment]
Thank you so much for the tips! I will give it a try and provide feedback here when I found where the issue is. Thanks!
